packages/features/bookings/lib/handleSeats.ts

```diff
@@ -0,0 +1,796 @@
+              metadata.conferenceData = updatedEvent.conferenceData;
+              metadata.entryPoints = updatedEvent.entryPoints;
+              evt.appsStatus = handleAppsStatus(results, newBooking, reqAppsStatus);
+            }
+          }
+        }
+
+        if (noEmail !== true && isConfirmedByDefault) {
+          const copyEvent = cloneDeep(evt);
+          loggerWithEventDetails.debug("Emails: Sending reschedule emails - handleSeats");
+          await sendRescheduledEmails({
+            ...copyEvent,
+            additionalNotes, // Resets back to the additionalNote input and not the override value
+            cancellationReason: `$RCH$${rescheduleReason ? rescheduleReason : ""}`, // Removable code prefix to differentiate cancellation from rescheduling for email
+          });
+        }
+        const foundBooking = await findBookingQuery(newBooking.id);
+
+        resultBooking = { ...foundBooking, appsStatus: newBooking.appsStatus };
+      } else {
+        // Merge two bookings together
+        const attendeesToMove = [],
+          attendeesToDelete = [];
+
+        for (const attendee of booking.attendees) {
+          // If the attendee already exists on the new booking then delete the attendee record of the old booking
+          if (
+            newTimeSlotBooking.attendees.some(
+              (newBookingAttendee) => newBookingAttendee.email === attendee.email
+            )
+          ) {
+            attendeesToDelete.push(attendee.id);
+            // If the attendee does not exist on the new booking then move that attendee record to the new booking
+          } else {
+            attendeesToMove.push({ id: attendee.id, seatReferenceId: attendee.bookingSeat?.id });
+          }
+        }
+
+        // Confirm that the new event will have enough available seats
+        if (
+          !eventType.seatsPerTimeSlot ||
+          attendeesToMove.length +
+            newTimeSlotBooking.attendees.filter((attendee) => attendee.bookingSeat).length >
+            eventType.seatsPerTimeSlot
+        ) {
+          throw new HttpError({ statusCode: 409, message: "Booking does not have enough available seats" });
+        }
+
+        const moveAttendeeCalls = [];
+        for (const attendeeToMove of attendeesToMove) {
+          moveAttendeeCalls.push(
+            prisma.attendee.update({
+              where: {
+                id: attendeeToMove.id,
+              },
+              data: {
+                bookingId: newTimeSlotBooking.id,
+                bookingSeat: {
+                  upsert: {
+                    create: {
+                      referenceUid: uuid(),
+                      bookingId: newTimeSlotBooking.id,
+                    },
+                    update: {
+                      bookingId: newTimeSlotBooking.id,
+                    },
+                  },
+                },
+              },
+            })
+          );
+        }
+
+        await Promise.all([
```

Comment:
I wonder if we could use prisma transactions here...

Reply:
Fixed in https://github.com/calcom/cal.com/pull/12905/commits/adddd1bb77c5134bca19ae244bcfc9b49c025a15